lib/ui/platform_dispatcher.dart

```diff
@@ -801,11 +801,41 @@ class PlatformDispatcher {
   ///
   ///  * [SchedulerBinding], the Flutter framework class which manages the
   ///    scheduling of frames.
+  ///  * [imposeSyncFrame], a similar method that is used in rare cases that
+  ///    a frame must be rendered immediately.
   void scheduleFrame() => _scheduleFrame();
 
   @Native<Void Function()>(symbol: 'PlatformConfigurationNativeApi::ScheduleFrame')
   external static void _scheduleFrame();
 
+  /// Immediately render a frame by invoking the [onBeginFrame] and
+  /// [onDrawFrame] callbacks synchronously.
+  ///
+  /// This method performs the same computation for a frame as [scheduleFrame]
+  /// does, but instead of doing so at an appropriate opportunity, the render is
+  /// completed synchronously within this call.
+  ///
+  /// This method should not be used in usual cases. It is designed for
+  /// situations that require a frame is rendered as soon as possible, even at
+  /// the cost of rendering quality.
+  ///
+  /// See also:
+  ///
+  ///  * [SchedulerBinding], the Flutter framework class which manages the
+  ///    scheduling of frames.
+  ///  * [scheduleFrame].
+  ///
+  // TODO(dkwingsmt): This method is not used for now, and is not implemented by
+  // the engine yet. This is because we have to land the Dart FFI method first
+  // before being able to run the performance test for the full changes due to
+  // the restriction of the performance test. Eventually, we should either land
+  // the full changes, or remove this method.
+  // https://github.com/flutter/flutter/issues/142851
+  void imposeSyncFrame() => _imposeSyncFrame();
```

Comment:
In terms of naming, I favor `forceSyncFrame`. I do like the Sync in the name, though.

Reply:
Why not just call it warmUpFrame? or something like that.

Reply:
> Why not just call it warmUpFrame? or something like that.

I think "warm up frame" is more application-level and belongs to the service binding, and for `PlatformDispatcher` it's better to name it with what the developer makes the engine do.

Reply:
but its only used for a warm up frame, and probably isn't usefulfor anything besides a warm up frame.

Reply:
We're expecting some developers calling the `dart:ui` package directly instead of going through the framework, who will have their own scheduling scheme and terms. I think it's better to leave room for them. Just like, in `scheduleFrame`, we're using very vague terms like "at appropriate opportunity" instead of directly pointing out "at the next vsync", because different platforms can have different schemes of frame scheduling.

Reply:
But if you don't know what the usages are then its unlikely that you'll be able to design it in such a way as to accomidate them.

Reply:
I've added "used in `scheduleWarmUpFrame`" to document. Does it feel better?